cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java

```diff
@@ -370,6 +372,38 @@ public void testSearchMetrics() throws Exception {
                                                       "system.writes", "system.zz", "user.reads", "user.writes"));
   }
 
+  @Test
+  public void testResultLimit() throws Exception {
+    long start = 1;
+    Map<String, String> sliceBy = getFlowletContext("resolutions", "WordCount1", "WordCounter", "run1", "splitter");
+
+    // 1 second
+    metricStore.add(new MetricValue(sliceBy, "reads", start, 1, MetricType.COUNTER));
+    // 30 second
+    metricStore.add(new MetricValue(sliceBy, "reads", start + 30, 1, MetricType.COUNTER));
+    // 1 minute
+    metricStore.add(new MetricValue(sliceBy, "reads", start + 60, 1, MetricType.COUNTER));
+    // 10 minutes
+    metricStore.add(new MetricValue(sliceBy, "reads", start + 600, 1, MetricType.COUNTER));
+    // 1 hour
+    metricStore.add(new MetricValue(sliceBy, "reads", start + 3600, 1, MetricType.COUNTER));
+
+    // just one record
```

Comment:
put in comment that you test:
* limit is 1
* limit is less than data points in time range
* limit is greater than data points in time range